src/radix_tree.cpp

```diff
@@ -0,0 +1,583 @@
+/*
+    Copyright (c) 2018 Contributors as noted in the AUTHORS file
+
+    This file is part of libzmq, the ZeroMQ core engine in C++.
+
+    libzmq is free software; you can redistribute it and/or modify it under
+    the terms of the GNU Lesser General Public License (LGPL) as published
+    by the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    As a special exception, the Contributors give you permission to link
+    this library with independent modules to produce an executable,
+    regardless of the license terms of these independent modules, and to
+    copy and distribute the resulting executable under terms of your choice,
+    provided that you also meet, for each linked independent module, the
+    terms and conditions of the license of that module. An independent
+    module is a module which is not derived from or based on this library.
+    If you modify this library, you must extend this exception to your
+    version of the library.
+
+    libzmq is distributed in the hope that it will be useful, but WITHOUT
+    ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+    FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public
+    License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "precompiled.hpp"
+#include "macros.hpp"
+#include "err.hpp"
+#include "radix_tree.hpp"
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+node::node(unsigned char* data)
+    : data_(data)
+{}
+
+uint32_t node::refcount()
+{
+    uint32_t u32;
+    memcpy(&u32, data_, sizeof(u32));
+    return u32;
+}
+
+void node::set_refcount(uint32_t value)
+{
+    memcpy(data_, &value, sizeof(value));
+}
+
+uint32_t node::prefix_length()
+{
+    uint32_t u32;
+    memcpy(&u32, data_ + sizeof(uint32_t), sizeof(u32));
+    return u32;
+}
+
+void node::set_prefix_length(uint32_t value)
+{
+    memcpy(data_ + sizeof(value), &value, sizeof(value));
```

Comment:
Too complicated. Function call is overkill.
The code could be this instead:

`*(reinterpret_cast<uint32_t *>(data_) + 1) = value;`